client/shared/src/api/contract.ts

```diff
@@ -45,6 +46,14 @@ export interface MainThreadAPI {
      */
     applySettingsEdit: (edit: SettingsEdit) => Promise<void>
 
+    /**
+     * GraphQL request API
+     */
+    requestGraphQL: <TResult extends object = any, TVariables extends object = any>(
+        request: string,
+        variables: TVariables
+    ) => Promise<GraphQLResult<TResult>>
```

Comment:
I would suggest we just not use generics in this interface given they always disappear anyway. That way we don't need to cast elsewhere.

```suggestion
    requestGraphQL: (request: string, variables: any) => Promise<GraphQLResult<any>>
```